src/test/java/htsjdk/tribble/annotation/StrandTest.java

```diff
@@ -0,0 +1,113 @@
+package htsjdk.tribble.annotation;
+
+import htsjdk.HtsjdkTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Unit tests for {@link Strand}.
+ * <p>
+ *     Makes sure that {@link Strand} will behave according to contract in run-time.
+ * </p>
+ */
+public class StrandTest extends HtsjdkTest {
+
+    @Test
+    public void testEncodeReturnNoNulls() {
+        for (final Strand subject : Strand.values()) {
+            Assert.assertNotNull(subject.encode());
+        }
+    }
+
+    @Test
+    public void testEncodeDecode() {
+        for (final Strand subject : Strand.values()) {
+            final String encode = subject.encode();
+            final Strand back = Strand.decode(encode);
+            Assert.assertSame(back, subject);
+        }
+    }
+
+    @Test
+    public void testEncodeDecodeAsChar() {
+        for (final Strand subject : Strand.values()) {
+            final char encode = subject.encodeAsChar();
+            final Strand back = Strand.decode(encode);
+            Assert.assertSame(back, subject);
+        }
+    }
+
+    @Test
+    public void testEncodeUniqueness() {
+        Assert.assertEquals(Stream.of(Strand.values())
+                .map(Strand::encode)
+                .collect(Collectors.toSet()).size(), Strand.values().length);
+    }
+
+    @Test
+    public void testEncodeAsCharUniqueness() {
+        Assert.assertEquals(Stream.of(Strand.values())
+                .map(Strand::encodeAsChar)
+                .collect(Collectors.toSet()).size(), Strand.values().length);
+    }
+
+    @Test
+    public void testEncodeAsStringAndCharAreEquivalent() {
+        for (final Strand subject : Strand.values()) {
+            final String str = subject.encode();
+            final char ch = subject.encodeAsChar();
+            Assert.assertEquals(str, String.valueOf(ch));
+        }
+    }
+
+    @Test
+    public void testAliases() {
+        Assert.assertEquals(Strand.FORWARD, Strand.POSITIVE);
+        Assert.assertEquals(Strand.REVERSE, Strand.NEGATIVE);
+    }
+
+    @Test
+    public void testToStrandMatchesDecode() {
+        for (final Strand value : Strand.values()) {
+            Assert.assertSame(Strand.toStrand(value.encode()), Strand.decode(value.encode()));
+        }
+    }
+
+    @Test
+    public void testToStringMatchesEncode() {
+        for (final Strand value : Strand.values()) {
+            Assert.assertEquals(value.toString(), value.encode());
+        }
+    }
```

Comment:
it would be god to have an explicit test that shows that you can decode valid chars and strings into the three difference `Strand`s.